cub/util_type.cuh

```diff
@@ -39,12 +39,13 @@
 #include <limits>
 #include <type_traits>
 
-#if (__CUDACC_VER_MAJOR__ >= 9 || CUDA_VERSION >= 9000) && !_NVHPC_CUDA
+#include <cuda.h>
+
+#if !_NVHPC_CUDA
     #include <cuda_fp16.h>
 #endif
-#if (__CUDACC_VER_MAJOR__ >= 11 || CUDA_VERSION >= 11000) && !_NVHPC_CUDA &&   \
-  !defined(CUB_DISABLE_BF16_SUPPORT)
-#include <cuda_bf16.h>
+#if !_NVHPC_CUDA && !defined(CUB_DISABLE_BF16_SUPPORT)
```

Comment:
Does bf16 not work on nvc++?

If so, would it make sense to just add the `_NVHPC_CUDA` check into `CUB_DISABLE_BF16_SUPPORT`?